js_modules/dagit/packages/core/src/ui/Loading.tsx

```diff
@@ -14,7 +14,17 @@ const BLANK_LOADING_DELAY_MSEC = 500;
 
 export const Loading = <TData extends Record<string, any>>(props: ILoadingProps<TData>) => {
   const {children, purpose, allowStaleData = false, renderError} = props;
-  const {error, data, loading} = props.queryResult;
+  const {error, data: _data, loading} = props.queryResult;
```

Comment:
I think the issue might actually be due to `useCursorPaginatedQuery` not returning stale data. so maybe this logic should actually live inside `useCursorPaginatedQuery`. Basically because `useCursorPaginatedQuery` never returns stale data when the variables change (might be intentional? I'm not sure) then the `allowStaleData` flag here doesn't do anything since we never give it stale data.